lib/koans/18_genservers.ex

```diff
@@ -0,0 +1,68 @@
+defmodule GenServers do
+  use Koans
+
+  @intro "GenServers"
+
+  defmodule ExampleGenServer do
+    use GenServer
+
+    # Use init to set up initial state []
+    def init(_args) do
+      {:ok, []}
+    end
+
+    # Use calls to perform synchronous actions
+    def handle_call(:pop, _from, [head|tail]) do
+      {:reply, head, tail}
+    end
+
+    def handle_call(:peek, _from, stack) do
+      {:reply, stack, stack}
+    end
+
+    # Use casts to perform asynchronous actions
+    def handle_cast({:push, value}, stack) do
+      {:noreply, [value|stack]}
+    end
+  end
+
+
+  koan "You can think of GenServers like more flexible agents" do
```

Comment:
As far I remember, I don't think the interface is actually _used_ by supervisors, however, their interfaces especially in terms of initialisation have matching specs, which makes them easy to handle together.

I do agree that the comparison to agents is probably not the best idea, though. I wanted to hint previous koan-sets with that but agree that there's way better comparisons.